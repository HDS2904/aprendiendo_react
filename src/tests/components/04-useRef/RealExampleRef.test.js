import { fireEvent, render } from "@testing-library/react"
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef"

describe('Prueba en <RealExampleRef />>', () => {
	let wrapper
	beforeEach( () => {
		wrapper = render( <RealExampleRef /> )
	})

	test('Debe mostrarse correctamente', () => {

		// const wrapper = render( <RealExampleRef /> )
		expect( wrapper.container ).toMatchSnapshot()
		expect( wrapper.queryByText(  'BreakingBad Quotes' ) ).toBe( null )

	})

	test('Debe de mostrar el componente <MultipleCustomHooks />', () => {
		// const wrapper = render( <RealExampleRef /> )
		const button = wrapper.getByRole('button', { name: /Ocultar/i });
		fireEvent.click(button)
		// wrapper.queryByText( 'BreakingBad Quotes' ) // prueba suficiente para saber si existe
		expect( !!wrapper.queryByText( 'BreakingBad Quotes' ) ).toBe( true )
		
	})

})