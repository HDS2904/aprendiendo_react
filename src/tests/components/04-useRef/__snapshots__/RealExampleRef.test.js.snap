// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <RealExampleRef />> Debe mostrarse correctamente 1`] = `
<div>
  <div>
    <h1>
      RealExampleRef
    </h1>
    <hr />
    <button
      class="btn btn-primary mt-5"
    >
      Ocultar
    </button>
  </div>
</div>
`;
